File: ws/src/my_py_pkg/my_py_pkg/my_node.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class MyNode(Node):
    def __init__(self):
        super().__init__('my_py_node')
        self.get_logger().info("Hello from Python node!")

def main(args=None):
    rclpy.init(args=args)
    node = MyNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: ws/src/my_cpp_pkg/src/my_node.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

int main(int argc, char **argv) {
  rclcpp::init(argc, argv);
  auto node = rclcpp::Node::make_shared("my_cpp_node");
  RCLCPP_INFO(node->get_logger(), "Hello from C++ node!");
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}
